packages/zoltan2/src/directory/Zoltan2_Directory_Impl.hpp

```diff
@@ -595,8 +538,7 @@ int Zoltan2_Directory<gid_t,lid_t,user_t>::update_local(
   node.owner = owner;
   node.errcheck = owner;
 
-  if(!node_map.insert({*gid, node}).second) {
-  // if(node_map.insert(*gid, node).failed()) { // TODO Switch to Kokkos map
+  if(node_map.insert(*gid, node).failed()) {
     // Need more nodes (that's the assumption here if we have failure)
```

Comment:
`failed()` isn't the only possible non-successful result; see `freed_existing` in `Kokkos_UnorderedMap.hpp`.

Reply:
I think freed_existing() is only relevant for parallel or threaded cases? I can add this when we develop parallel forms.